src/screens/title.lua

```diff
@@ -14,15 +16,30 @@ function TitleScreen (state)
                 color_index = 1
             end
         end
+        
         game_over_timer = 0
-        -- Move to character select screen
-        if btnp(4) then
-            state.screen = screens.character_select
-    
-            -- Reset the timer and score upon starting the game
-            timer = 0
-            timer_frame_counter = 0
-            state.score = 0
+        
+        -- Navigate menu
+        if btnp(2) then -- up
+            selection = selection - 1
+            if selection < 1 then selection = #menu_items end
+        end
+        if btnp(3) then -- down
+            selection = selection + 1
+            if selection > #menu_items then selection = 1 end
+        end
+        
+        -- Select menu option
+        if btnp(4) then -- 'Z'
+            if selection == 1 then
+                state.screen = screens.character_select
+                -- Reset the timer and score upon starting the game
+                timer = 0
+                timer_frame_counter = 0
+                state.score = 0
+            elseif selection == 2 then
+                state.screen = screens.credits
```

Comment:
This will pull the persistent instance of the credits screen. I haven't properly managed this. as a result the credits never reset when you go back.

change this line to state.screen = Credits(state)

we technically only need to do this on stateful screens that should reset on load